src/main/java/org/llorllale/cactoos/matchers/Throws.java

```diff
@@ -48,13 +50,27 @@
  * @param <T> Type of the scalar's value
  * @since 1.0.0
  * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ * @checkstyle MemberNameCheck (200 lines)
+ * @checkstyle ParameterNameCheck (200 lines)
+ * @todo #108:30min Refactor 'Throws' to eliminate fields with compound names.
+ *  There are two fields with compound names in this class:
+ *  'messageExpectation' and 'expectationDescription'. Compound names indicate
+ *  that the scope is too large and so cohesion too low. Create a new nested
+ *  class, maybe 'MessageExpectation', and move these fields and related
+ *  logic to this class.
  */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.LongVariable"})
 public final class Throws<T> extends TypeSafeDiagnosingMatcher<Scalar<T>> {
 
     /**
-     * The expected exception message.
+     * The expectation which the exception message must match.
      */
-    private final String msg;
+    private final UncheckedFunc<String, Boolean> messageExpectation;
+
+    /**
+     * The expectation which the exception message must match.
+     */
+    private final String expectationDescription;
```

Comment:
@vzurauskas Although debatable, but those projects [deprecate compound words as variables](https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html), use, just `description` or just `expectation`

Reply:
@ilyakharlamov I agree that compound names are not good, however, I think just changing them to `description` and `expectation` would eliminate the symptom but not the problem. The problem is the scope is too large, making single-word names ambiguous. If a variable was named `description`, it wouldn't be crear a description of what: of exception, of messagge, of matcher? Similarly, if a variable was named `expectation`, it wouldn't be clear an expectation of what: exception type or exception message?

I think one way to solve the problem of too large scope is to create a new object, `MessageExpectation` (or something along those lines) moving these two fields (together with their logic) into it. Then the scope would become small enough for them to have single-word names. I wrote a `@todo` for that.

Reply:
@vzurauskas the refactor should be for `Throws` to extend `MatcherEnvelope` - see #120 

I don't see enough rewards for the additional investment in a new `MessageExpectation`. By extending `MatcherEnvelope`, all variables should be in a small and tight scope that should be easier to digest. 

There are serious problems with both these new variables `messageExpectation` and `expectationDescription`:
* javadocs on both is exactly the same
* "expectation" is a prefix in one and a suffix in another (let's have it as a prefix)
* "expectation" should just be "expected"

Reply:
@llorllale I don't see a good way to fit `Throws` into `MatcherEnvelope` because filling out `Description` has to happen in the same subroutine as exception catching - description depends on whether exception was thrown or not. In `Throws` currenctly `Description` is filled out in `matchesSafely`, but `MatcherEnvelope`'s constructor takes matching and describing functions separately.

My mistake about the javadocs, fixed.

We can't have `expectation` as prefix in both. We have `messageExpectation`, which is the expectation for the exception message (the rule we expect the message to conform to), and `expectationDescription`, which is the description of this expectation. If we made `expectation` a prefix for both, the expectation for exception message would be called what, `expectationMessage`? That wouldn't make sense, because that variable represents the expectation for the message and not the message itself. Renaming `expectation` to `expected` wouldn't help either, because then it would become `expectedMessage`, which still is message and not expectation for the message.

So I see only two ways of making these variable names unumbiguous: leaving them as they are now, but then they would stay too long, or shortening  them to one word, but then they would need smaller scope. Or perhaps giving them some completely different names, but I can't think up any.